lib/pacemaker/pcmk_options.c

```diff
@@ -38,3 +40,23 @@ pcmk__list_cluster_options(pcmk__output_t *out)
     return pcmk__output_cluster_options(out, name, desc_short, desc_long,
                                         pcmk__opt_context_none);
 }
+
+// Documented in header
+int
+pcmk_list_cluster_options(xmlNode **xml)
```

Comment:
This could take the place of the tmp_out usage

Reply:
You'd think so :/ But this doesn't take a filter argument (needed by `pcmk__daemon_metadata()`) or a set of name/desc_long/desc_short arguments (needed by `fencer_metadata()`).